keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -45,7 +46,11 @@ public BayesianNetwork(Collection<? extends Vertex> vertices) {
     public double getLogOfMasterP() {
         double sum = 0.0;
         for (Vertex<?> vertex : latentAndObservedVertices) {
-            sum += vertex.logProbAtValue();
+            if (vertex instanceof Probabilistic) {
+                sum += ((Probabilistic)vertex).logProbAtValue();
+            } else if (vertex.isObserved() && !vertex.matchesObservation()) {
```

Comment:
AFAIK Keanu isn't currently designed to allow observations on Non Probabilistic Vertices (see the implementation of observe() in the class NonProbabilisticDouble for example).  It looks like these diffs assume that they are at the moment so perhaps we should have a chat to discuss the correct approach moving forwards?

Reply:
Albeit - it does appear that you *can* observe a non-probabilistic Bool in the current version of Keanu...

Reply:
Yes, and in particular if you remove this line it breaks `MetropolisHastingsTest#samplesFromPriorWithObservedDeterministic`